src/libraries/Microsoft.PowerFx.Interpreter/Environment/Governor.cs

```diff
@@ -0,0 +1,48 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Diagnostics.Contracts;
+using System.Globalization;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.PowerFx.Core.Entities;
+using Microsoft.PowerFx.Core.IR;
+using Microsoft.PowerFx.Core.IR.Nodes;
+using Microsoft.PowerFx.Core.IR.Symbols;
+using Microsoft.PowerFx.Core.Texl.Builtins;
+using Microsoft.PowerFx.Functions;
+using Microsoft.PowerFx.Interpreter;
+using Microsoft.PowerFx.Types;
+using static Microsoft.PowerFx.Functions.Library;
+
+namespace Microsoft.PowerFx
+{
+    /// <summary>
+    /// Throttling object to allow a host to control limits of memory and hardware resources.
+    /// These virtuals are called by the evaluator, and exceptions thrown by them will abort the evaluation. 
+    /// This is a service passed to <see cref="RuntimeConfig"/>, and a derived class can override and throw their own host exceptions if hardware limits are exceeded.
+    /// </summary>
+    public class Governor
```

Comment:
>Governor [](http://example.com/codeflow?start=17&length=8)

We want to define if this class will work for Sync and/or Async Power Fx APIs

When a governor is used, we should validate if it's compatible with the API that is being used.